### PR TITLE
Resolve some compiler warnings

### PR DESCRIPTION
**Pull request by rmackay9**
Travis produces a fair number of compiler warnings some of which are resolved below.  Resolving these will make the output a bit more readable.

### CODE REVIEW

**Comment by OXINARF** [comment]
This is great Randy! If you have a wish to look to more warnings look at Clang builds :smile:

File: libraries/AP_AccelCal/AP_AccelCal.cpp [hunk 1]
```diff
@@ -56,7 +56,10 @@ void AP_AccelCal::update()
             case ACCEL_CAL_WAITING_FOR_ORIENTATION: {
                 // if we're waiting for orientation, first ensure that all calibrators are on the same step
                 uint8_t step;
-                cal = get_calibrator(0);
+                if ((cal = get_calibrator(0)) == NULL) {
```
Comment (magicrub) [inline libraries/AP_AccelCal/AP_AccelCal.cpp]: Tsk tsk, I frown upon  assignments within an if statement.

Comment (rmackay9) [inline libraries/AP_AccelCal/AP_AccelCal.cpp]: Yeah, fair enough, I'll fix that.  I actually did it the normal way first.. and doing it this way made me think.. hmm.. is this ok?

**Comment by rmackay9** [comment]
as a side note matrix_alg seems to have a mess of problems according to coverity.

**Comment by OXINARF** [comment]
@rmackay9 I can't access the page, can you add me please?

**Comment by rmackay9** [comment]
@OXINARF, invite sent to telepac address.

**Comment by OXINARF** [comment]
@rmackay9 Thanks!

**Comment by lucasdemarchi** [comment]
> as a side note matrix_alg seems to have some problems according to coverity.

yep... both L_inv and U_inv are leaking. The amount of small allocations in these functions are perturbing.

File: APMrover2/commands_logic.cpp [hunk 2]
```diff
@@ -186,7 +186,7 @@ void Rover::do_nav_wp(const AP_Mission::Mission_Command& cmd)
     // this will be used to remember the time in millis after we reach or pass the WP.
     loiter_time = 0;
     // this is the delay, stored in seconds
-    loiter_time_max = abs(cmd.p1);
```
Comment (magicrub) [inline APMrover2/commands_logic.cpp]: is this better to clip at zero instead?

File: libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp [hunk 3]
```diff
@@ -506,7 +506,7 @@ void AP_Frsky_Telem::calc_gps_position()
         
         alt = loc.alt * 0.01f;
         _alt_gps_meters = (int16_t)alt;
-        _alt_gps_cm = (alt - abs(_alt_gps_meters)) * 100;
```
Comment (magicrub) [inline libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp]: good catch, this was probably a bug

**Comment by magicrub** [comment]
merged with a minor change of "loiter_time_max = (cmd.p1 >= 1) ? cmd.p1 : 0;"

**Comment by magicrub** [comment]
doh! my bad, I thought that was a float. I reverted my "fix" back to what you had.

**Pull request closed without merging**